packages/datasources/snowflake/index.cjs

```diff
@@ -381,5 +382,47 @@ module.exports.options = {
 				}
 			}
 		}
+	},
+	proxy: {
+		title: 'Connect through an authenticated proxy?',
+		type: 'boolean',
+		secret: false,
+		required: false,
+		nest: true,
+		children: {
+			[true]: {
+				host: {
+					title: 'Host',
+					type: 'string',
+					secret: false,
+					required: true
+				},
+				port: {
+					title: 'Port',
+					type: 'number',
+					secret: false,
+					required: false
```

Comment:
maybe this should be required, but I would assume there is a default port.

Reply:
I'm not entirely sure I follow, but there isn't a default specified here, does the snowflake library provide a default when this is used?

Reply:
ahh nvm if you pass hostname port is also required. This is the error log they spit out `[ ! ] Error connecting to datasource data_warehouse: A proxyPort must be specified.`

Reply:
here is the connector code for checking proxy params https://github.com/snowflakedb/snowflake-connector-nodejs/blob/cba00c1848f875b8e2054cf485bd9f507a7eee9c/lib/connection/connection_config.js#L213-L271